src/gui/DatabaseTabWidget.cpp

```diff
@@ -157,63 +154,22 @@ void DatabaseTabWidget::openDatabase(const QString& fileName, const QString& pw,
     }
     file.close();
 
-    QLockFile* lockFile = new QLockFile(QString("%1/.%2.lock").arg(fileInfo.canonicalPath(), fileInfo.fileName()));
-    lockFile->setStaleLockTime(0);
-
-    if (!dbStruct.readOnly && !lockFile->tryLock()) {
-        // for now silently ignore if we can't create a lock file
-        // due to lack of permissions
-        if (lockFile->error() != QLockFile::PermissionError) {
-            QMessageBox msgBox;
-            msgBox.setWindowTitle(tr("Database already opened"));
-            msgBox.setText(tr("The database you are trying to open is locked by another instance of KeePassXC.\n\n"
-                              "Do you want to open it anyway?"));
-            msgBox.setIcon(QMessageBox::Question);
-            msgBox.addButton(QMessageBox::Yes);
-            msgBox.addButton(QMessageBox::No);
-            auto readOnlyButton = msgBox.addButton(tr("Open read-only"), QMessageBox::NoRole);
-            msgBox.setDefaultButton(readOnlyButton);
-            msgBox.setEscapeButton(QMessageBox::No);
-            auto result = msgBox.exec();
-
-            if (msgBox.clickedButton() == readOnlyButton) {
-                dbStruct.readOnly = true;
-                delete lockFile;
-                lockFile = nullptr;
-            } else if (result == QMessageBox::Yes) {
-                // take over the lock file if possible
-                if (lockFile->removeStaleLockFile()) {
-                    lockFile->tryLock();
-                }
-            } else {
-                delete lockFile;
-                return;
-            }
-        }
-    }
-
     Database* db = new Database();
     dbStruct.dbWidget = new DatabaseWidget(db, this);
-    dbStruct.lockFile = lockFile;
-    dbStruct.saveToFilename = !dbStruct.readOnly;
-
-    dbStruct.filePath = fileInfo.absoluteFilePath();
-    dbStruct.canonicalFilePath = canonicalFilePath;
-    dbStruct.fileName = fileInfo.fileName();
+    dbStruct.fileInfo = fileInfo;
 
     insertDatabase(db, dbStruct);
 
     if (dbStruct.readOnly) {
         emit messageTab(tr("File opened in read only mode."), MessageWidget::Warning);
     }
 
-    updateLastDatabases(dbStruct.filePath);
+    updateLastDatabases(dbStruct.fileInfo.absoluteFilePath());
 
     if (!(pw.isNull() && keyFile.isEmpty())) {
```

Comment:
Please reformulate this condition to `!pw.isNull() || !keyFile.isEmpty()`, it's a lot easier to comprehend.